src/luascript.h

```diff
@@ -299,24 +299,24 @@ class LuaScriptInterface
 		static typename std::enable_if<std::is_integral<T>::value && std::is_unsigned<T>::value, T>::type
 			getNumber(lua_State* L, int32_t arg)
 		{
-			LUA_NUMBER luaNum = lua_tonumber(L, arg);
-			if (luaNum < 0 || luaNum > std::numeric_limits<T>::max()) {
-				reportErrorFunc(L, fmt::format("Passed argument '{:s}' has invalid value: {:d}", arg, luaNum));
+			double num = lua_tonumber(L, arg);
+			if (num < 0.0 || num > static_cast<double>(std::numeric_limits<T>::max())) {
+				reportErrorFunc(L, fmt::format("Passed argument '{}' has invalid value: {}", arg, num));
 			}
 
-			return static_cast<T>(luaNum);
+			return static_cast<T>(num);
 		}
 
 		template<typename T>
-		static typename std::enable_if<(std::is_integral<T>::value && std::is_signed<T>::value) || std::is_floating_point<T>::value, T>::type
+		static typename std::enable_if<(std::is_integral<T>::value && (std::is_signed<T>::value) || std::is_floating_point<T>::value), T>::type
 			getNumber(lua_State* L, int32_t arg)
 		{
-			LUA_NUMBER luaNum = lua_tonumber(L, arg);
-			if (luaNum > std::numeric_limits<T>::max()) {
-				reportErrorFunc(L, fmt::format("Passed argument '{:s}' has invalid value: {:d}", arg, luaNum));
+			double num = lua_tonumber(L, arg);
+			if (num > static_cast<double>(std::numeric_limits<T>::max())) {
```

Comment:
Shouldn't you be checking for min value too?
```suggestion
			if (num > static_cast<double>(std::numeric_limits<T>::max()) || num < static_cast<double>(std::numeric_limits<T>::min())) {
```
Or something like that?

Reply:
I was, but it broke for some reason xD

Reply:
What broke?

Reply:
```
Lua Script Error: [Spell Interface]
data/spells/scripts/support/haste.lua
LuaScriptInterface::getNumber(). Passed argument '5' has invalid value: -24
stack traceback:
        [C]: in function 'setFormula'
        data/spells/scripts/support/haste.lua:7: in main chunk

Lua Script Error: [Spell Interface]
data/spells/scripts/support/haste.lua
LuaScriptInterface::getNumber(). Passed argument '3' has invalid value: -24
stack traceback:
        [C]: in function 'setFormula'
        data/spells/scripts/support/haste.lua:7: in main chunk

Lua Script Error: [Test Interface]
data/spells/scripts/support/sharpshooter.lua
LuaScriptInterface::getNumber(). Passed argument '4' has invalid value: -0.7
stack traceback:
        [C]: in function 'setFormula'
        data/spells/scripts/support/sharpshooter.lua:14: in main chunk
```